csrc/optimization/opt_pass.cpp

```diff
@@ -0,0 +1,119 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <optimization/opt_pass.h>
+
+#include <mutex>
+
+namespace nvfuser::optimization {
+
+namespace {
+
+// TODO: throw away toy flags. Will redo this at later stage
+thread_local bool pre_segmenter_flag = true;
+
+void setOptimizationFlag(const OptimizationPassCategory& flag, bool enabled) {
+  if (flag == OptimizationPassCategory::PreSegmenter) {
+    pre_segmenter_flag = enabled;
+  }
+}
+
+bool getOptimizationFlag(const OptimizationPassCategory& flag) {
+  if (flag == OptimizationPassCategory::PreSegmenter) {
+    return pre_segmenter_flag;
+  }
+  return false;
+}
+
+class OptimizationRegistry {
+ public:
+  struct PassEntry {
+    int priority_;
+    FusionPass pass_;
+    std::string name_;
+    PassEntry(int priority, FusionPass pass, std::string name)
+        : priority_(priority), pass_(std::move(pass)), name_(std::move(name)) {}
+  };
+
+  void registerPass(
+      const OptimizationPassCategory& cat,
+      FusionPass func,
+      std::string name_,
+      int priority) {
+    std::lock_guard<std::mutex> guard(mutex_);
+    auto& pass_entry_list = pass_categories_[cat];
+    auto entry_iter = pass_entry_list.begin();
+    while (entry_iter != pass_entry_list.end()) {
```

Comment:
It looks like we are trying to order the pass by its priority and store to `unordered_map`, can this be done by simpley using ordered `map`?

Reply:
I was actually using `list`. The key for unordered_map is barely just a grouping mechanism. priority is used to set their order within the group.
we would definitely want to rework the whole manager & grouping mechanics. So I'd only view current code as a placeholder.